### PR TITLE
Sort menu redesign - Bottom Sheet Dialog

### LINKED ISSUE #1039: Arranging the form names in ascending/descending order provides bad user experience.
#### Software and hardware versions 
Collectv1.6.1

#### Problem description
Arranging the forms in ascending or descending order on the basis of their names opens a NavDrawer from right side.

#### Steps to reproduce the problem
Instead of using a NavDrawer <A-Z> and <Z-A> button can be used.

#### Expected behavior
It works properly but has a bad user experience.

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #1039

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/utilities/ApplicationConstants.java [hunk 1]
```diff
@@ -28,6 +31,24 @@
             "sq", "sw", "sw_KE", "ta", "th_TH", "tl", "tr", "uk", "ur",
             "ur_PK", "vi", "zh", "zu"};
 
+    public static int getSortIcon(String sortBy) {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/utilities/ApplicationConstants.java]: This constants class should really only hold true constants and no logic.

**Review by lognaturel — DISMISSED** [review]
This looks and works really well for me in portrait! It disappears on screen rotation, though, and doesn't show all the way in landscape. This is on a Galaxy Tab A, Android 5.1.1.

**Comment by shobhitagarwal1612** [comment]
@lognaturel Please review

**Comment by lognaturel** [comment]
Nice, @shobhitagarwal1612! For Edit Saved Form, there are a lot of options and so some scrolling is needed to see all of them. That wasn't immediately obvious to me. Is there some standard way of identifying that scrolling may be needed?

@mmarciniak90 can you please do a test pass? @grzesiek2010 would be great to get your eyes on it as well and see what you think. I also want to do another code review pass.

**Comment by mmarciniak90** [comment]
I noticed problems. Tested on Android 4.1, 4.2, 4.4, 5.1.
Use case:
1. User is on 'Get Blank forms' view
2. User opens sort dialog
3. User changes device orientation and backs to vertical orientation
4. User locks screen 
5. User unlocks screen
6. User sees that sort dialog is opened second time

Other similar case:
1. User is on 'Get Blank forms' view
2. User opens sort dialog
3. User changes device orientation and backs to vertical orientation
4. User closes sort dialog
5. User locks screen 
6. User unlocks screen
7. Sort dialog appears

Those problems occur only on sort list with 2 options

**Comment by shobhitagarwal1612** [comment]
@mmarciniak90 @lognaturel Please review

**Comment by lognaturel** [comment]
@grzesiek2010 @mmarciniak90 @yanokwa I think this is really nice and I think it's pretty unlikely to confuse users. What would you think about merging it to `master`? That way we'd have a solution that better matches current design guidelines and that could stay the same through a more significant UI refresh.

**Comment by grzesiek2010** [comment]
Samsung Galaxy Duos
Android 4.2.2

> 07-25 11:13:26.009 15937-15937/org.odk.collect.android E/WindowManager: android.view.WindowLeaked: Activity org.odk.collect.android.activities.InstanceChooserList has leaked window com.android.internal.policy.impl.PhoneWindow$DecorView{c29124d V.E..... R....... 0,0-1920,1005} that was originally added here
                                                                            at android.view.ViewRootImpl.<init>(ViewRootImpl.java:465)
                                                                            at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:277)
                                                                            at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:69)
                                                                            at android.app.Dialog.show(Dialog.java:312)
                                                                            at org.odk.collect.android.activities.AppListActivity.setupBottomSheet(AppListActivity.java:313)
                                                                            at org.odk.collect.android.activities.AppListActivity.onResume(AppListActivity.java:136)
                                                                            at org.odk.collect.android.activities.InstanceChooserList.onResume(InstanceChooserList.java:156)
                                                                            at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1255)
                                                                            at android.app.Activity.performResume(Activity.java:6412)
                                                                            at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3392)
                                                                            at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3434)
                                                                            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2772)
                                                                            at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4471)
                                                                            at android.app.ActivityThread.access$1000(ActivityThread.java:177)
                                                                            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1455)
                                                                            at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                            at android.os.Looper.loop(Looper.java:145)
                                                                            at android.app.ActivityThread.main(ActivityThread.java:5951)
                                                                            at java.lang.reflect.Method.invoke(Native Method)
                                                                            at java.lang.reflect.Method.invoke(Method.java:372)
                                                                            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
                                                                            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)

I see this crash if i open the sort menu and change the device orientation. it works well only in `Send Finalized Form` and `Delete Saved Form`

I'm not able to reproduce this issue on Samsung Galaxy S4 Android 5.0.1

**Comment by grzesiek2010** [comment]
Both icons are blue if i select one of available options (it work well by default after installation):
screenshot_2017-07-25-15-32-30
screenshot_2017-07-25-15-32-37

I noticed this issue also using `Samsung Galaxy Duos Android 4.2.2` it looks ok on `Samsung Galaxy S4 Android 5.0.1`

**Comment by shobhitagarwal1612** [comment]
@grzesiek2010 Fixed

**Comment by grzesiek2010** [comment]
ok I'll look at it again.

**Comment by shobhitagarwal1612** [comment]
@lognaturel I have switched the base branch to master

**Comment by grzesiek2010** [comment]
The crash doesn't occur but the problem with blue icons is still visible.

**Comment by shobhitagarwal1612** [comment]
@grzesiek2010 In which screen did you notice this?

**Comment by grzesiek2010** [comment]
each one

**Comment by shobhitagarwal1612** [comment]
@grzesiek2010 I am unable to reproduce the bug. I am testing on API 25 emulator. Can you please describe the steps to reproduce this bug?

**Comment by grzesiek2010** [comment]
There aren't any special steps just:
1. Open any list
2. Open Sort menu
3. Click on Name, A-Z or Name, Z-A
5. Open the menu again and both icons (Name, A-Z or Name, Z-A) are blue.

The same if you use Date options. Everything is ok if you sort by status because there are two different icons I guess.

**Comment by shobhitagarwal1612** [comment]
@grzesiek2010 Fixed. Please test again.

**Comment by grzesiek2010** [comment]
@shobhitagarwal1612 @lognaturel I can confirm that both issues I found have been fixed. @mmarciniak90 will continue testing when she is free.

**Comment by lognaturel** [comment]
@shobhitagarwal1612 If you fix these merge conflicts one more time, this is ready to go in. 🎉
 @mmarciniak90 can take another pass at it after it's merged and file issues if needed.

**Comment by shobhitagarwal1612** [comment]
@lognaturel Merged the conflicts

**Review by lognaturel — APPROVED** [review]

**Pull request merged**